### Comparativa entre PCA por Matriz de Covarianza y Correlación

In [3]:
Sigma <- matrix(c(1, 4,
                  4, 100), nrow = 2, byrow = TRUE)

In [4]:
sd_vec <- sqrt(diag(Sigma))
R_mat <- diag(1/ sd_vec) %*% Sigma %*% diag(1/ sd_vec)
round(R_mat, 6)

[,1] [,2]
[1,] 1.0  0.4 
[2,] 0.4  1.0

In [5]:
eig_sigma <- eigen(Sigma)
eig_sigma

eigen() decomposition
$values
[1] 100.1613532   0.8386468

$vectors
           [,1]        [,2]
[1,] 0.04030552 -0.99918740
[2,] 0.99918740  0.04030552


In [6]:
eig_corr <- eigen(R_mat)
eig_corr

eigen() decomposition
$values
[1] 1.4 0.6

$vectors
          [,1]       [,2]
[1,] 0.7071068 -0.7071068
[2,] 0.7071068  0.7071068


Las componentes principales de la matriz de correlación son:
* $y_1 = 0.7071*z_1 + 0.7071*z_2$
* $y_2 = -0.7071*z_1 + 0.7071*z_2$

Mientras que los componentes principales de la matriz de covarianza son:
* $y_1 = 0.0403*x_1 + 0.9991*x_2$
* $y_1 = - 0.9991*x_1 + 0.0403*x_2$

Notando que $z_i$ son las variables estandarizadas y $x_i$ las variables originales 

### Comparativa en las Varianzas Explicadas

In [ ]:
# Varianza explicada por Matriz de Covarianza
eig_sigma$values / sum(eig_sigma$values)

[1] 0.991696566 0.008303434

In [8]:
# Varianza explicada por Matriz de Correlación
eig_corr$values / sum(eig_corr$values)

[1] 0.7 0.3

Usando la matriz de covarianza se tiene que solo se requiere de un componente para explicar los datos. Mientras que de la de correlación se requiere de ambos componentes para interpretar y explicar los datos

### Correlaciones entre Componentes y Variables

In [14]:
# Correlaciones en Matriz de Covarianza y_i , x_k

rho_YX <- matrix(NA,2,2,
dimnames = list(paste0("Y",1:2),paste0("X",1:2)))

for (i in 1:2) 
{
    for (j in 1:2)
    {
        rho_YX[i,j] <- eig_sigma$vectors[j,i] * sqrt(eig_sigma$values[i]/Sigma[j,j])
    }
}

rho_YX

X1         X2         
Y1  0.4033802 0.999993188
Y2 -0.9150325 0.003691085

In [17]:
# Correlaciones en Matriz de Correlación y_i , z_k

rho_YZ <- matrix(NA,2,2,
dimnames = list(paste0("Y",1:2),paste0("Z",1:2)))

for (i in 1:2) 
{
    for (j in 1:2)
    {
        rho_YZ[i,j] <- eig_corr$vectors[j,i] * sqrt(eig_corr$values[i])
    }
}

rho_YZ

Z1         Z2       
Y1  0.8366600 0.8366600
Y2 -0.5477226 0.5477226

In [21]:
round(t(eig_corr$vectors) %*% R_mat %*% eig_corr$vectors,2)

[,1] [,2]
[1,] 1.4  0.0 
[2,] 0.0  0.6

In [22]:
eig_corr$values

[1] 1.4 0.6